docs/book/core-concepts.md

```diff
@@ -104,7 +104,9 @@ Artifacts can be serialized and deserialized (i.e. written and read from the Art
 
 **Materializers**
 
-A materializer defines how and where Artifacts live in between steps.
+A materializer defines how and where Artifacts live in between steps. It is used to convert a ZenML artifact into a specific format. They are most often used to handle the input or output of ZenML steps, and can be extended by building on the `BaseMaterializer` class. We care about this because steps are not just isolated pieces of work; they are linked together and the outputs of one step might well be the inputs of the next.
+
+We have some built-in ways to serialize and deserialize the data flowing between steps. Of course, if you are using some library or tool which doesn't work with our built-in options, you can write [your own custom materializer](https://docs.zenml.io/guides/low-level-api/chapter-5) to ensure that your data can be passed from step to step in this way.
```

Comment:
Maybe write here about the fileio role in this. I.e using the fileio we ensure it works across artifact stores?